packages/components/src/Modal.ts

```diff
@@ -0,0 +1,245 @@
+import { CSSProperties, ReactNode, useEffect, useRef } from 'react';
+import { div, h } from 'react-hyperscript-helpers';
+import RModal, { Props as RModalProps } from 'react-modal';
+
+import { ButtonPrimary, ButtonSecondary } from './buttons';
+import { Clickable } from './Clickable';
+import { useUniqueId } from './hooks/useUniqueId';
+import { icon } from './icon';
+import { getPopupRoot } from './internal/PopupPortal';
+
+export const modalStyles = {
+  overlay: {
+    position: 'fixed',
+    left: 0,
+    right: 0,
+    top: 0,
+    bottom: 0,
+    overflowY: 'auto',
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'flex-start',
+    padding: '2rem 1rem',
+    backgroundColor: 'rgba(0, 0, 0, 0.5)',
+  },
+  modal: {
+    position: 'relative',
+    padding: '1.5rem 1.25rem',
+    borderRadius: 5,
+    backgroundColor: 'white',
+    boxShadow: '0 0 8px 0 rgba(0,0,0,0.12), 0 8px 8px 0 rgba(0,0,0,0.24)',
+    outline: 'none',
+  },
+  header: {
+    display: 'flex',
+    alignItems: 'baseline',
+    flex: 'none',
+    marginBottom: '1rem',
+  },
+  title: {
+    overflow: 'hidden',
+    fontSize: 18,
+    fontWeight: 600,
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+  },
+  buttonRow: {
+    display: 'flex',
+    justifyContent: 'flex-end',
+    alignItems: 'baseline',
+    marginTop: '1rem',
+  },
+} as const satisfies Record<string, CSSProperties>;
```

Comment:
almost missed this as was gonna suggest the type above... any reason why we need to use const satisfies pattern instead of just using the type above near the declaration?

Reply:
With ` as const`, the keys are type checked and accessing a key that doesn't exist produces a type error.

If the variable is typed as `Record<string, CSSProperties>`, TS allows accessing keys that don't exist.

Reply:
ah, right.  yeah, I remember now.  thx!